### ISSUE TITLE
Clean unable to drop function in db2

### ISSUE THREAD

**Issue by msallstr** [flyway/flyway#1154]
I'm getting the following db2 error when trying to run clean
```bash
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:3.2.1:clean (default) on project uppfoljning: org.flywaydb.core.api.FlywayException: Unable to drop "TEST_FUNC(CHARACTER,CHARACTER): DB2 SQL Error: SQLCODE=-458, SQLSTATE=42883, SQLERRMC=UPPFOLJNING.TEST_FUNC, DRIVER=4.19.26 -> [Help 1]
```

From the [db2 documentation](https://www-01.ibm.com/support/knowledgecenter/SSEPEK_10.0.0/com.ibm.db2z10.doc.codes/src/tpc/n458.html)
_For example, assuming a function exists for which a parameter was defined as CHAR(12) on the CREATE FUNCTION statement, a reference to that function by a signature could specify this parameter as either CHAR(12), or CHAR(). The CHAR() syntax provides a way to say, "do not care about length, precision and scale attributes in finding a matching function"._

A simple test case
```sql
CREATE FUNCTION TEST_FUNC ( PARAM1 char(16), PARAM2 char(36) ) 
   RETURNS INTEGER 
   LANGUAGE SQL 
RETURN 1; 

drop function test_func(character, character); -- fails
drop function test_func(character(), character()); -- succeeds
drop function test_func(character(16), character(36)); -- succeeds
```

To never care about length may be the way to fix this.

Using flywaydb version 3.2.1 and DB2 v10.5.500.107 with fixpack 5

**Comment by pswrdf** [comment]
DB2 v9.7 Flyway doesn't drops a function on clean operation (from java):

CREATE FUNCTION db2admin.get_version()
  RETURNS VARCHAR(16)
  SPECIFIC "GET_VERSION"
  NO EXTERNAL action F1:
  BEGIN atomic
    RETURN '7.0.0';
  END 

console show no errors:
1:44:06 PM org.flywaydb.core.internal.command.DbClean cleanSchema
INFO: Cleaned schema "DB2ADMIN" (execution time 00:00.016s).

But fails on followed by migrate operation: Found non-empty schema "DB2ADMIN" without metadata table!
And the function still exists in schema.

**Comment by axelfontaine** [comment]
Fixed thanks to #1155 by @umefjord
